rclcpp/test/rclcpp/test_publisher.cpp

```diff
@@ -199,7 +199,7 @@ TEST_F(TestPublisher, test_publisher_with_system_default_qos) {
   using test_msgs::msg::Empty;
   ASSERT_NO_THROW(
   {
-    auto publisher = node->create_publisher<Empty>("topic", rclcpp::SystemDefaultsQoS());
+    auto publisher = node->create_publisher<Empty>("topic", rclcpp::SystemDefaultsQoS(), options);
```

Comment:
NodeOption disables intra-process communication in the context, and it should use publisher option to enable it explicitly but never used for the publisher.